slsim/Sources/point_plus_extended_source.py

```diff
@@ -0,0 +1,83 @@
+import numpy.random as random
+from slsim.Sources.source_pop_base import SourcePopBase
+import numpy as np
+from slsim.selection import deflector_cut
+from slsim.Sources.galaxies import galaxy_projected_eccentricity
+
+
+class PESource(SourcePopBase):
+    """Class to describe point and extended sources."""
+
+    def __init__(
+        self,
+        pes_list,
+        cosmo,
+        sky_area,
+        kwargs_cut,
+        variability_model=None,
+        kwargs_variability_model=None,
+    ):
+        """
+
+        :param pes_list: list of dictionary with point and extended source parameters
+        :param cosmo: cosmology
+        :type cosmo: ~astropy.cosmology class
+        :param sky_area: Sky area over which galaxies are sampled. Must be in units of
+            solid angle.
+        :type sky_area: `~astropy.units.Quantity`
+        :param variability_model: keyword for the variability model to be used. This is
+         a population argument, not the light curve parameter for the individual
+         source.
+        :param kwargs_variability_model: keyword arguments for the variability of
+         a source. This is a population argument, not the light curve parameter for
+         the individual source.
+        """
+        super(PESource, self).__init__(
+            cosmo=cosmo,
+            sky_area=sky_area,
+            variability_model=variability_model,
+            kwargs_variability_model=kwargs_variability_model,
+        )
+        self.n = len(pes_list)
+        ## check missing kewords in astropy table.
+        column_names = pes_list.colnames
+        if "ellipticity" not in column_names:
```

Comment:
these lines of code are perhaps very similar/duplicate to the one in the extended source class. Perhaps we can mere them, or even have this class inherited from the extended source class?